presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -817,6 +817,9 @@ private PhysicalOperation visitScanFilterAndProject(
                     return new PhysicalOperation(operatorFactory, outputMappings, source);
                 }
             }
+            catch (PrestoException pe) {
```

Comment:
This fix seems correct, but I'd like to understand what cases this covers because categorized errors already seem to work properly. Can you put this in a separate commit with a test case that fails without this?